tests/test_grafana.py

```diff
@@ -0,0 +1,27 @@
+"""Tests for the Grafana containers."""
+
+import pytest
+import requests
+import tenacity
+from pytest_container.container import ContainerData
+
+from bci_tester.data import GRAFANA_CONTAINERS
+
+
+@pytest.mark.parametrize("container", GRAFANA_CONTAINERS, indirect=True)
+def test_prometheus_healthy(container: ContainerData) -> None:
+    """Simple smoke test verifying that Grafana is healthy."""
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential()
+    )
+    def _fetch_grafana_health() -> requests.Response:
+        port = container.forwarded_ports[0].host_port
+        return requests.get(f"http://localhost:{port}/api/health", timeout=2)
```

Comment:
This might not work, as tenacity will only retry on raised exceptions. However a 400/404/500 response from grafana will not be treated as a failure. For that to happen, you have to tweak the code slightly:
```suggestion
    @tenacity.retry(
        stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential()
    )
    def _fetch_grafana_health() -> requests.Response:
        port = container.forwarded_ports[0].host_port
        resp = requests.get(f"http://localhost:{port}/api/health", timeout=2)
        resp.raise_for_status()
        return resp
```

If you pick this code, then you can also drop the status code assertion a few lines below.

Reply:
This is no longer necessary as of https://github.com/SUSE/BCI-dockerfile-generator/pull/1306